recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class MongoCxxConan(ConanFile):
+    name = "mongo-cxx-driver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://mongocxx.org"
+    description = "C++ Driver for MongoDB"
+    topics = ("conan", "libbsoncxx", "libmongocxx", "mongo", "mongodb", "database", "db")
+    settings = "os", "compiler", "arch", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = ("cmake", "cmake_find_package")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "polyfill": ["std", "boost", "mnmlstc", "experimental"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "polyfill": "boost"
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio" and self.options.polyfill != "boost":
+            raise ConanInvalidConfiguration("For MSVC, best to use the boost polyfill")
```

Comment:
Don't raise in this case, default is to use C++17 if supported: https://github.com/mongodb/mongo-cxx-driver/blob/5704837c094f322df7bf4ef32b0809655ae47b95/src/bsoncxx/CMakeLists.txt#L40, so I guess it works with msvc.

But you can raise, whatever the compiler, if `polyfill == mnmlstc` because it is not yet available in CCI.